### PR TITLE
chore: node init auto fetches mainnet genesis, new config/app defaults

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

When someone wants to init a new osmosisd node, they start by running `osmosisd init NAME`

The normal process for me after doing that is:
1. Change config.toml to set snapshot interval to 0
2. Download the osmosis genesis file and replace the temp one that gets generated

This PR automates this process. If the chain ID is blank or osmosis-1, we fetch the osmosis mainnet genesis file. If the chainID is anything else, we generate a new one like the previous behavior. If the node doesn't have internet connection and/or cant fetch the genesis for whatever reason, we gracefully continue and generate a default genesis just like before.

## Testing and Verifying

Locally tested:

1. Init with no chainID fetches mainnet genesis
2. Init with osmosis-1 fetches mainnet genesis
3. Initi with anything else generates the normal default genesis
4. Init with no chainID or osmosis-1 with no internet connection generates the normal default genesis

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

## Summary by CodeRabbit

- **New Features**
	- Introduced Osmosis-specific defaults to the node initialization process, enhancing the configuration for various modules like P2P, Mempool, and more.
	- Added functionality for automatic genesis file generation and downloading.

### CODE REVIEW

File: cmd/osmosisd/cmd/init.go [hunk 1]
```diff
@@ -136,53 +146,82 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
...
+			tmcfg.WriteConfigFile(filepath.Join(config.RootDir, "config", "config.toml"), config)
 			err = writeConfigToFile(filepath.Join(config.RootDir, "config", "client.toml"), nil)
 			if err != nil {
 				return errors.Wrap(err, "Failed to write client.toml file")
 			}
 
+			// Create .env file if FlagSetEnv is true
 			createEnv, _ := cmd.Flags().GetBool(FlagSetEnv)
 			if createEnv {
 				err = CreateEnvFile(cmd)
```
[the remark this answers is not in the document]
Comment (czarcas7ic) [reply cmd/osmosisd/cmd/init.go:228]: Bug was actually there prior to this PR, but nice find https://github.com/osmosis-labs/osmosis/pull/7795/commits/b1995015d9c918a783179beb10545584c7b2d4fd

File: cmd/osmosisd/cmd/init.go [hunk 2]
```diff
@@ -136,53 +141,82 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
...
+
+			if isMainnet {
+				// If the chainID is blank or osmosis-1, prep this as a mainnet node
+
+				// Attempt to download the genesis file from the Osmosis GitHub repository
+				// If fail, generate a new genesis file
+				err := downloadGenesis(config)
 				if err != nil {
-					return errors.Wrap(err, "Failed to read genesis doc from file")
+					fmt.Println("Failed to download genesis file, using default")
```
Comment (ValarDragon) [inline cmd/osmosisd/cmd/init.go:168]:
```suggestion
					fmt.Println("Failed to download genesis file, using a random chain ID and genesis file for local testing")
```

Comment (ValarDragon) [reply cmd/osmosisd/cmd/init.go:168]: Lets add a TODO indicating that this is a weird thing to do here, and maybe we should just fail?

**Review by ValarDragon — APPROVED** [review]
This LGTM, but I am a bit worried about confusion for integrators. Happy to try merging and see if we get issues for folks!


Ideally we can get this behavior into upstream SDK so its not going to be a custom tooling break for Osmosis. E.g. Asking in an issue if we can have apps provide a default chain ID / genesis file URL for init.

File: cmd/osmosisd/cmd/init.go (hunk 2 above)
Comment (czarcas7ic) [reply cmd/osmosisd/cmd/init.go:168]: This is actually the normal behavior of what happens right now, so its essentially defaulting to what it is doing now

Comment (czarcas7ic) [reply cmd/osmosisd/cmd/init.go:168]: But yeah happy to add a note on this

**Comment by czarcas7ic** [comment]
I am going to address the comments when I get time next week, and I might remove the timeout changes for a separate PR

**Pull request merged**